### DISCUSSION TITLE
berry cron -  no deterministic behavior

### DISCUSSION THREAD

**Discussion by wjohn007** [arendst/Tasmota#20696] [General]
I use the current stable Tasmota version 13.3.
Following code
```be
cronName ="test"
tasmota.remove_cron(cronName)
tasmota.add_cron("0-5 * * * * *", 
def () 
  print(tasmota.time_str(tasmota.rtc()["local"])) end
,"test")
```

Following output

```
2024-02-10T19:55:03
2024-02-10T19:56:00
2024-02-10T19:56:01
2024-02-10T19:56:04
2024-02-10T19:56:05
2024-02-10T19:57:02
2024-02-10T19:57:03
2024-02-10T19:58:00
2024-02-10T19:58:04
2024-02-10T19:58:05
2024-02-10T19:59:01
2024-02-10T19:59:02
2024-02-10T19:59:03
```

I would expect the anonymous method to run 6x per minute.

I am wrong ?

John

**Comment by sfromis** [comment]
To check if you get the expected 6 messages at the start of each minute, look in the Tasmota console instead of the Berry REPL console. The latter is known to be "less complete", but IDK exactly when. It may be related to other activity, and the polling done by the JavaScript code running in your browser. This may be affected by the tab being in the foreground or not, and maybe some plugin.

Anyway, running your test case for 10 minutes, I do see all 6 expected messages occurring stably, both in REPL and "classic" console, at a steady rate of one per second, without noteworthy "bumps" in the timing, This is as seen by the Tasmota console time stamps, specifically not by the visual appearance. In general, the web UI is not expected to be updated faster than every 2.345 seconds.

My test board was an ESP32-S3 without other activities running outside of standard Tasmota.

If reliable logging is important to you, the REPL console is not the place to do it. The standard Tasmota log facilities (including Weblog) are better for that, or you could consider creating a log file for your purpose.

**Comment by wjohn007** [comment]
Thank you for this fast answer.
I checked the console and all the expected messages are there.
Thanks for the hint.
